packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -3311,55 +3315,49 @@ export class ObjectsInRedisClient {
         }
     }
 
-    getObjectsByPatternAsync(pattern: string, options: CallOptions) {
+    getObjectsByPatternAsync(pattern: string, options: CallOptions): Promise<ioBroker.Object[] | void> {
         return new Promise((resolve, reject) =>
             this.getObjectsByPattern(pattern, options, (err, objs) => (err ? reject(err) : resolve(objs)))
         );
     }
 
-    async _setObject(
+    private async _setObject(
         id: string,
         obj: ioBroker.SettableObject,
-        options: CallOptions,
-        callback?: ioBroker.SetObjectCallback
-    ) {
+        options: CallOptions
+    ): ioBroker.SetObjectPromise {
```

Comment:
Like above, you might want to impose some restrictions on the type of `obj` based on the ID that's passed. See
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/06f38fe1967d69ed164552b55674bf2d5269e7a1/types/iobroker/index.d.ts#L696-L713